services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/RouterProcessor.java

```diff
@@ -36,13 +41,24 @@ private void getRouterSubnetIds(PortContext context, String vpcId, String subnet
                 fetchRouterSubnetsRequest, this::fetchConnectedSubnetIdsCallback);
     }
 
-    private void getRouterSubnetIds(PortContext context, List<PortEntity> portEntities) {
+    private void getRouterSubnetIds(PortContext context, List<PortEntity> portEntities) throws Exception {
         Set<String> vpcIds = new HashSet<>();
+        int tryTimes = TRY_TIMES;
         for (PortEntity portEntity: portEntities) {
             if (portEntity.getFixedIps() == null) {
                 continue;
             }
 
+            //Waiting for random ip address to be assigned
+            while (portEntity.getFixedIps() == null && tryTimes > 0) {
+                Thread.sleep(SLEEP_TIME);
```

Comment:
Could we make the TRY_TIMES and SLEEP_TIME configurable?

Reply:
This may be a temporary implementation, I am considering whether there is a better way to deal with request dependencies.

Reply:
I think we could have a generic retry mechanism across the processes, but each process could have its own set of configurations.

Reply:
I'm thinking of solving this problem through the observer design pattern, instead of blocking and waiting for dependent process completion in an asynchronous way.

Reply:
these configurations may not be needed in the future..

Reply:
Cool. Observer pattern is used in the case of one-to-many relationship between objects such as if one object is modified, its dependent objects are to be notified automatically. Does this apply to our scenario as you intend to use the asynchronous notification. 

This temp configuration is fine for now. We can merge this PR of current form to master, and you can take the time to design and leverage some advanced design pattern 👍